### PR TITLE
Add new Role and RoleBinding for sink.

### PR DESCRIPTION
**Pull request by ggallen**
Add new Role and RoleBinding for sink that allows it to delete resources in a namespace watched by `KubeArchive`

Resolves https://github.com/kubearchive/kubearchive/issues/219.

### CODE REVIEW

File: cmd/operator/internal/controller/kubearchiveconfig_controller.go [hunk 1]
```diff
@@ -202,51 +214,52 @@ func (r *KubeArchiveConfigReconciler) desiredRole(ctx context.Context, kaconfig
...
 			Kind:     "Role",
-			Name:     kaconfig.Name,
+			Name:     name,
 		},
 		Subjects: []rbacv1.Subject{{
 			Kind:      "ServiceAccount",
-			Name:      kaconfig.Name,
+			Name:      name,
 			Namespace: kaconfig.Namespace,
 		}},
```
Comment (skoved) [inline cmd/operator/internal/controller/kubearchiveconfig_controller.go:264]: The ServiceAccount for the sink is the kubearchive namespace, not in the same namespace as the KubeArchiveConfig

**Review by skoved — CHANGES_REQUESTED** [review]

File: cmd/operator/internal/controller/kubearchiveconfig_controller.go (hunk 1 above)
Comment (ggallen) [reply cmd/operator/internal/controller/kubearchiveconfig_controller.go:264]: Ah, good catch!

File: cmd/operator/internal/controller/kubearchiveconfig_controller.go [hunk 2]
```diff
@@ -202,52 +214,53 @@ func (r *KubeArchiveConfigReconciler) desiredRole(ctx context.Context, kaconfig
...
-		log.Error(err, "Failed to reconcile RoleBinding")
+		log.Error(err, "Failed to reconcile RoleBinding "+name)
 		return binding, err
 	}
 
 	return binding, nil
 }
 
-func (r *KubeArchiveConfigReconciler) desiredRoleBinding(kaconfig *kubearchivev1alpha1.KubeArchiveConfig) (*rbacv1.RoleBinding, error) {
+func (r *KubeArchiveConfigReconciler) desiredRoleBinding(kaconfig *kubearchivev1alpha1.KubeArchiveConfig, name string, namespace string) (*rbacv1.RoleBinding, error) {
```
Comment (skoved) [inline cmd/operator/internal/controller/kubearchiveconfig_controller.go:249]: you should add comment somewhere explaining that the `namespace` arg is specifically for the namespace of the Subject

**Review by skoved — APPROVED** [review]
other this one small thing, LGTM :+1:

**Review by skoved — APPROVED** [review]

**Pull request merged**